src/frontend/screens/MapScreen/BGMapSelector.tsx

```diff
@@ -0,0 +1,231 @@
+import Button from "../../sharedComponents/Button";
+import { useNavigation } from "react-navigation-hooks";
+import LocationContext from "../../context/LocationContext";
+
+const m = defineMessages({
+  title: {
+    id: "sharedComponents.BGMapSelector.title",
+    defaultMessage: "Background Maps",
+    description: "Title for the background map selector",
+  },
+  close: {
+    id: "sharedComponents.BGMapSelector.close",
+    defaultMessage: "Close",
+  },
+  manageMaps: {
+    id: "sharedComponents.BGMapSelector.manageMaps",
+    defaultMessage: "Manage Maps",
+  },
+});
+
+// To do: We should get this all from one central place
+// Perhaps as a Seperate Module published on npm
+interface mapServerStyle {
+  id: string;
+  styleUrl: string;
+  title?: string;
+}
+interface MapSelectorProps {
+  /** Should NOT come from `useBottomSheet()` */
+  closeSheet: () => void;
+  onMapSelected: (styleURL: string) => void;
+}
+
+/** `ref` should NOT come from - `useBottomSheet()` */
+export const BGMapSelector = React.forwardRef<
+  BottomSheetMethods,
+  MapSelectorProps
+>(({ closeSheet, onMapSelected }, ref) => {
+  const [bgMapsList, setBgMapList] = React.useState<null | mapServerStyle[]>(
+    null
+  );
+
+  const { navigate } = useNavigation();
+
+  const [snapPoints, setSnapPoints] = React.useState<(number | string)[]>([
+    0,
+    "40%",
+  ]);
+
+  const { formatMessage: t } = useIntl();
+
+  React.useEffect(() => {
+    // To do: Api call to get list styles
+    async function getListStyles(): Promise<mapServerStyle[]> {
+      return [
+        {
+          id: "121",
+          styleUrl: MapboxGL.StyleURL.Street,
+        },
+        {
+          id: "7aj1",
+          styleUrl: MapboxGL.StyleURL.Street,
+        },
+        {
+          id: "as121",
+          styleUrl: MapboxGL.StyleURL.Street,
+        },
+        {
+          id: "7afaj1",
+          styleUrl: MapboxGL.StyleURL.Street,
+        },
+        {
+          id: "12eagg1",
+          styleUrl: MapboxGL.StyleURL.Street,
+        },
+        {
+          id: "7af323j1",
+          styleUrl: MapboxGL.StyleURL.Street,
+        },
+      ];
+    }
+
+    getListStyles().then(stylesList => {
+      setBgMapList(stylesList);
+    });
+  }, []);
+
+  return (
+    <BottomSheet
+      ref={ref}
+      snapPoints={snapPoints}
+      backdropComponent={BottomSheetBackdrop}
+      enableContentPanningGesture={false}
+      enableHandlePanningGesture={false}
+      handleHeight={0}
+      handleComponent={() => null}
+    >
+      <View
+        onLayout={e => {
+          const { height } = e.nativeEvent.layout;
+          setSnapPoints([0, height]);
+        }}
+        style={{ padding: 20 }}
+      >
+        {bgMapsList === null ? (
+          <Loading />
+        ) : (
+          <View style={{ backgroundColor: WHITE }}>
+            <Text style={styles.title}> {t(m.title)}</Text>
+            <TouchableOpacity
+              activeOpacity={0.99}
+              onPress={() => {
```

Comment:
There is a slight opacity issue here where you can slightly see background map. Open to suggestions on how to solve this, but not a big deal. You can only see it if you are really looking!